Sources/SwiftDiagnostics/FixIt.swift

```diff
@@ -27,15 +27,45 @@ public protocol FixItMessage: Sendable {
   var fixItID: MessageID { get }
 }
 
+public protocol ReplacingChildData: Sendable {
+  associatedtype Parent: SyntaxProtocol
+  associatedtype Child: SyntaxProtocol
+
+  var replacementRange: Range<AbsolutePosition> { get }
+  var parent: Parent { get }
+  var newChild: Child { get }
+}
+
 /// A Fix-It that can be applied to resolve a diagnostic.
 public struct FixIt: Sendable {
   public enum Change: Sendable {
+    struct ReplacingOptionalChildData<Parent: SyntaxProtocol, Child: SyntaxProtocol>: ReplacingChildData {
+      let parent: Parent
+      let newChild: Child
+      let keyPath: WritableKeyPath<Parent, Child?> & Sendable
+
+      var replacementRange: Range<AbsolutePosition> {
+        if let oldChild = parent[keyPath: keyPath] {
+          return oldChild.range
+        } else {
+          let newParent = parent.with(keyPath, newChild)
+          if let previousToken = newParent[keyPath: keyPath]?.previousToken(viewMode: .all) {
+            return previousToken.endPosition..<previousToken.endPosition
+          } else {
+            return parent.position..<parent.position
+          }
```

Comment:
Is it necessary to use `previousToken().endPosition`? Can’t you use `newParent[keyPath: keyPath].position`?

Reply:
Agree, but should we add more test cases for this new API?

Reply:
I think the existing test in `PeerMacroTests` that generates a `replaceChild` Fix-It should be sufficient.